crates/mdb/src/lib.rs

```diff
@@ -21,124 +20,64 @@ macro_rules! suppress_unwind {
     };
 }
 
-type OnMessage = dyn FnMut(&Message) + Send + 'static;
-type OnError = dyn FnMut(&Error) + Send + 'static;
-type OnDone = dyn FnMut(Option<&Error>) + Send + 'static;
-
-unsafe fn pchar_to_string(p_value: *const c_char) -> String {
-    assert!(!p_value.is_null());
-    let value = String::from(CStr::from_ptr(p_value).to_str().unwrap());
-    value
-}
-
-enum OwnedOrBorrowedError {
-    Owned(*mut mdb_sys::mdb_error_t),
-    Borrowed(*const mdb_sys::mdb_error_t),
-}
-pub struct Error(OwnedOrBorrowedError);
-
-impl Debug for Error {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(f, "Code: {}; Message: {:?};", self.code(), self.message())
-    }
-}
+pub trait Callback<T>: FnMut(T) + Send + 'static {}
```

Comment:
Interesting, I have not seen this pattern for handling callbacks before. What are some benefits of using our own trait instead of relying on only the `Fn*` traits?

Have you seen any other crates or authors using it (i.e. is there anywhere I can learn more about it)?